src/main/scala/org/zalando/spearheads/innkeeper/api/JsonProtocols.scala

```diff
@@ -62,7 +63,22 @@ object JsonProtocols extends DefaultJsonProtocol {
     }
   }
 
-  implicit val endpointFormat = jsonFormat4(Endpoint.apply)
+  val endpointFormat = jsonFormat(Endpoint.apply, "hostname", "port", "protocol", "type")
+
+  implicit object EndpointFormat extends JsonFormat[Endpoint] {
+    override def write(obj: Endpoint): JsValue = endpointFormat.write(obj)
+
+    override def read(json: JsValue): Endpoint = {
+      val endpoint = endpointFormat.read(json)
+
+      Endpoint(
+        hostname = endpoint.hostname,
+        port = endpoint.port.orElse(Some(443)),
+        protocol = endpoint.protocol.orElse(Some(Https)),
+        endpointType = endpoint.endpointType.orElse(Some(ReverseProxy))
```

Comment:
I'm kind of unhappy repeating the defaults here and in the case class again. Isn't it enough to have them here? You are copying the endpoint anyway and apply the defaults here if there are none, right?